index.js

```diff
@@ -0,0 +1,135 @@
+const searchInput = document.getElementById("search__input");
+const searchButton = document.getElementById("search__btn");
+const cityLocation = document.getElementById("location");
+const weatherDescription = document.getElementById("description");
+const locationError = document.querySelector(".location__error");
+const weatherbox = document.getElementById("weather__box");
+const temprature = document.getElementById("temp");
+const loader = document.getElementById("loader");
+const weatherImg = document.getElementById("weatherImg");
+const errorMessage = document.getElementById("error");
+
+// API key and base URL for OpenWeatherMap API
+const API_KEY = "82005d27a116c2880c8f0fcb866998a0";
+const baseUrl = "http://api.openweathermap.org/data/2.5/weather?";
+
+// Function to update the UI with weather details
+const showWeatherDetails = (weatherinfo) => {
+  loader.classList.add("hidden");
+  weatherbox.classList.remove("hidden");
+  locationError.classList.add("hidden");
+  searchInput.value = "";
+  cityLocation.innerText = weatherinfo.name + " ," + weatherinfo.sys.country;
+  weatherDescription.innerText = weatherinfo.weather[0].description;
+  weatherImg.src = "./images/" + weatherinfo.weather[0].icon + ".png";
+  temprature.innerText = (weatherinfo.main.temp - 273.15).toFixed(1) + "°";
+  document.body.style.backgroundImage = `url(images/${weatherinfo.weather[0].main}.jpg)`;
+};
+
+const fetchDatafromCoordinates = async (data) => {
+  try {
+    const url = `${baseUrl}lat=${data.coord.lat}&lon=${data.coord.lon}&appid=${API_KEY}`;
+    await fetch(url)
+      .then((response) => response.json())
+      .then((data) => showWeatherDetails(data) 
+    );
+     
+  } catch (error) {
+    loader.classList.add("hidden");
+    weatherbox.classList.add("hidden");
+    locationError.classList.remove("hidden");
+    errorMessage.innerText = "Oops city not found.";
+  }
+};
+
+// Function to fetch weather details based on city name
+const getCityCoordinates = async () => {
+  const cityName = searchInput.value.trim();
+  loader.classList.remove("hidden");
+  weatherbox.classList.add("hidden");
+  locationError.classList.add("hidden");
+
+  if (cityName === "") {
+    loader.classList.add("hidden");
+    weatherbox.classList.add("hidden");
+    locationError.classList.add("hidden");
+    return;
+  }
+
+  const url = `${baseUrl}q=${cityName}&appid=${API_KEY}`;
+
+  // Fetch weather data from the API
+  await fetch(url)
```

Comment:
create a function which accepts url and gives response and call that function whenever you need don't repeat same code again and again

Reply:
toggleLoader(true)
toggleLoader = (state) => {
loader.classList.remove(state ? "show": "hidden");
  weatherbox.classList.add(state ? "show": "hidden");
  locationError.classList.add(state ? "show": "hidden");

}

Reply:
const cityName = searchInput.value.trim();
if(!cityName) {
return
}
logic......